pkg/client/introduce/client.go

```diff
@@ -52,22 +60,42 @@ func New(ctx Provider) (*Client, error) {
 		Event:   introduceSvc,
 		service: introduceSvc,
 		store:   store,
+		newUUID: func() string { return uuid.New().String() },
 	}, nil
 }
 
+// ProtocolDependency returns dependency for the service
```

Comment:
i think this is confusing in a developer facing API. i don't grok it from the description.

Reply:
note: we also need to allow for deferred continues in controller APIs (particularly REST). See DID exchange allowing for continue operations by connection ID.

Reply:
I have changed it a bit. ` the event I got from the client anyways. it is a bit surprising that this is needed.` there are two options: you can handle action event and pass your dependency (managing by yourself) or you can always use ProtocolDependency(msg.TreadID()) function.

Or we can add ProtocolDependency function to the service (like a service dependency) so, in this way, we can get rid of e.Continue(c.ProtocolDependency(thID)) bc service can call it by yourself

Reply:
I would like to pass it as a dependency function to the service. What do you think?